library/cwm/src/modules/CWM.rb

```diff
@@ -1061,11 +1065,10 @@ def validate_current_widgets(event)
     publish function: :InitNull, type: "void (string)"
     publish function: :StoreNull, type: "void (string, map)"
 
-  private
-
     def widgets_in_contents(contents)
       contents.each_with_object([]) do |arg, res|
         case arg
+        when ::CWM::CustomWidget then res.concat(arg.nested_widgets) << arg
```

Comment:
This method shouldn't do it. Instead, define AbstractWidget#widget_ids which by default returns just `[widget_id]` and the nested ones for CustomWidget.
```rb
if arg.respond_to? :widget_ids
  res.concat(arg.widget_ids)
else
  res.concat(widgets_in_contents(arg))
end